.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,163 @@
+name: unit_tests
+
+on: [push, pull_request]
+
+env:
+  NODE_VERSION: 10
+  SOLR_VERSION: 6.6.5
+
+jobs:
+
+  build-static:
+    name: Build static files
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+    - name: Cache node modules
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: npm-${{ hashFiles('package-lock.json') }}
+        restore-keys: |
+          npm-${{ hashFiles('package-lock.json') }}
+          npm-
+    - name: Install dependencies
+      run: npm ci
+    - name: Run webpack build
+      run: npm run build:prod
+    - name: Upload built files
+      uses: actions/upload-artifact@v2
+      with:
+        name: bundles
+        path: bundles/
+    - name: Upload stats file
+      uses: actions/upload-artifact@v2
+      with:
+        name: webpack-stats.json
+        path: webpack-stats.json
+
+  ts-unit:
+    name: TypeScript unit tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+    - name: Cache node modules
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: npm-${{ hashFiles('package-lock.json') }}
+        restore-keys: |
+          npm-${{ hashFiles('package-lock.json') }}
+          npm-
+    - name: Install dependencies
+      run: npm ci
+    - name: Run Jest
+      run: npx jest --collectCoverage
+    - name: Upload test coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        flags: typescript
+
+  py-unit:
+    name: Python unit tests
+    runs-on: ubuntu-latest
+    needs: build-static
+    services:
+      db:
+        image: mariadb
+        env:
+          MYSQL_DATABASE: sandco
+          MYSQL_ROOT_PASSWORD: sandco
+        ports:
+        - 3306:3306
+      solr:
+        image: solr:6.6
+        ports:
+          - 8983:8983
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    # use docker cp to copy the configset, then bash to set ownership to solr
+    - name: Copy Solr configset to solr service
+      run: |
+        docker cp solr_conf ${{ job.services.solr.id }}:/opt/solr/server/solr/configsets/sandco
+        docker exec --user root ${{ job.services.solr.id }} /bin/bash -c "chown -R solr:solr /opt/solr/server/solr/configsets/sandco"
```

Comment:
awesome that this worked; maybe we could extract it to a shell script at some point